docs/middleware.md

```diff
@@ -0,0 +1,73 @@
+# Middleware
+
+Use or release EdgeSpec-compatible middleware with a proper type-pipeline:
+
+```typescript
+import type { Middleware } from "edgespec"
+export const someMiddleware: Middleware<
+  // Middleware input
+  {
+    db: DatabaseClient
+  },
+  // Middleware output
+  {
+    is_authenticated: boolean
+  }
+> = (req) => {
+  const authToken = req.headers.get("authorization")?.split("Bearer ")?.[1]
+  if (!authToken) {
+    // EdgeSpec will attach returned properties to the Request object
+    return {
+      is_authenticated: false,
+    }
+  }
+
+  const [user] = req.db.query("SELECT * FROM users WHERE token=?", [authToken])
+
```

Comment:
Maybe show an example of overriding headers as well (see #31 i asked about how to handle this exactly)

Reply:
if this gets merged before #31 I'll let you update it in #31, otherwise I can update this with what we decide on

Reply:
trying to include as little speculative content as possible since that's been confusing people

Reply:
yeah we can come back to it dont worry about it